src/EnergyPlus/EarthTube.cc

```diff
@@ -405,23 +405,41 @@ namespace EarthTube {
 			}
 		}
 
-		// Check to make sure there is only on ventilation statement per zone
-		for ( Loop = 1; Loop <= TotEarthTube; ++Loop ) {
-			for ( Loop1 = Loop + 1; Loop1 <= TotEarthTube - 1; ++Loop1 ) {
-				if ( EarthTubeSys( Loop ).ZonePtr == EarthTubeSys( Loop1 ).ZonePtr ) {
-					ShowSevereError( cAlphaArgs( 1 ) + " is assigned to more than one " + cCurrentModuleObject );
-					ShowContinueError( "Only one such assignment is allowed." );
-					ErrorsFound = true;
-				}
-			}
-		}
+		CheckEarthTubesInZones( cAlphaArgs( 1 ), cCurrentModuleObject, ErrorsFound );
 
 		if ( ErrorsFound ) {
 			ShowFatalError( cCurrentModuleObject + ": Errors getting input.  Program terminates." );
 		}
 
 	}
 
+	
+	void
```

Comment:
:+1: for breaking up this functional section of code into a standalone function to allow for meaningful unit testing!